### PR TITLE
refac: logginscope for authenticated B2B actors

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
See the title.

## References
https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/229

## Checklist
- [x] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [ ] Subsystem test executed (dev_002/dev_003)
- [ ] Is there time to monitor state of the release to Production?
- [x] Reference to the task

### CODE REVIEW

File: source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs [hunk 1]
```diff
@@ -57,14 +57,13 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
...
         }
 
-        var serializer = context.InstanceServices.GetRequiredService<ISerializer>();
-        WriteAuthenticatedIdentityToLog(authenticatedActor.CurrentActorIdentity, serializer);
-        await next(context);
-    }
-
-    private void WriteAuthenticatedIdentityToLog(ActorIdentity? actorIdentity, ISerializer serializer)
-    {
-        _logger.LogInformation("Successfully authenticated market actor identity.");
```
Comment (ebbeknudsen) [inline source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs:67]: Should we still have this log statement to ensure  the authenticated user is atleast logged once?

Comment (lasrinnil) [reply source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs:67]: Yeah you are right I have reimplemented it, along with if the caller is not authenticated.

File: source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs [hunk 2]
```diff
@@ -50,21 +50,21 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
 
         var authenticated = await authenticationMethod.AuthenticateAsync(httpRequestData, context.CancellationToken);
 
-        if (!authenticated)
+        _logger.BeginScope(
+            "ActorNumber: {ActorNumber}, ActorRole: {ActorRole}, Restriction: {Restriction}",
+            authenticatedActor.CurrentActorIdentity.ActorNumber.Value,
+            authenticatedActor.CurrentActorIdentity.MarketRole?.Name,
+            authenticatedActor.CurrentActorIdentity.Restriction.Name);
```
Comment (ebbeknudsen) [inline source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs:57]: If `!authenticated` is it  still allowed to access ` authenticatedActor.CurrentActorIdentity`?

**Review by ebbeknudsen — APPROVED** [review]

File: source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs (hunk 2 above)
Comment (lasrinnil) [reply source/B2BApi/Configuration/Middleware/Authentication/MarketActorAuthenticatorMiddleware.cs:57]: No you are correct. I am reverting the change

**Review by MadsDue — APPROVED** [review]

**Pull request closed without merging**